packages/shared/src/types/config/dev.ts

```diff
@@ -76,6 +82,10 @@ export interface DevConfig {
    * Used to control whether the build artifacts of the development environment are written to the disk.
    */
   writeToDisk?: boolean | ((filename: string) => boolean);
+  /**
+   * This option allows you to configure a list of globs/directories/files to watch for file changes.
+   */
+  watchFiles?: string | string[] | WatchFiles;
```

Comment:
Can we simplify the typing to `WatchFiles`, i.e. remove the `string | string[]` types here.

The motivation is to make this config option easier to merge. When calling the `mergeRsbuildConfig` method, it is hard to merge a `string | string[] | object` type in the current implementation.

see: https://github.com/web-infra-dev/rsbuild/blob/2c397f083dda6b366936b5aa58114271f2062ec6/packages/core/src/mergeConfig.ts

Reply:
If we want to support the `string | string[]` types, we need to normalize the config to object format in the `mergeRsbuildConfig` method.

Reply:
> Can we simplify the typing to `WatchFiles`, i.e. remove the `string | string[]` types here.
> 
> The motivation is to make this config option easier to merge. When calling the `mergeRsbuildConfig` method, it is hard to merge a `string | string[] | object` type in the current implementation.
> 
> see: https://github.com/web-infra-dev/rsbuild/blob/2c397f083dda6b366936b5aa58114271f2062ec6/packages/core/src/mergeConfig.ts

Do you mean that?
```
watchFiles?: { paths: string[]; options?: WatchOptions; }
```

Reply:
Yes, this is a bit verbose, but good for merging.

Reply:
[8789c11](https://github.com/web-infra-dev/rsbuild/pull/2145/commits/8789c11bec38c8ddddd6368b30c62d54b9875e72)